### PR TITLE
[RFC][UN-8400] Variable pricing on enricher frontend

### PR DESCRIPTION
**Pull request by henrmota**
https://uniplaces.atlassian.net/browse/UN-8400

### CODE REVIEW

File: addon/components/uni-input-price.js [hunk 1]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
   type: 'number',
   options: null,
 
+  /**
+   * Converts a value to an integer when able, or leaves it as a string
+   * @public
+   */
+  numberAsString: computed('value', {
+    get() {
+      return this.get('value') === null ? '' : `${this.get('value')}`;
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]:
```js
return isPresent(this.get('value')) ? `${this.get('value')}` : '';
```
or
```js
return isEmpty(this.get('value')) ? '': `${this.get('value')}`;
```

File: addon/components/uni-input-price.js [hunk 2]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
   type: 'number',
   options: null,
 
+  /**
+   * Converts a value to an integer when able, or leaves it as a string
+   * @public
+   */
+  numberAsString: computed('value', {
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]: numberAsInt

File: addon/components/uni-input-price.js [hunk 3]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
...
+    get() {
+      return this.get('value') === null ? '' : `${this.get('value')}`;
+    },
+
+    set(_, value) {
+      if (this.get('type') === INPUT_TYPE_NUMBER) {
+        let newValue = this._isInt(value) ? parseInt(value) : value;
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger

File: addon/components/uni-input-price.js [hunk 4]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
...
+        this.set('value', newValue);
+
+        return newValue;
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]: Remove this return

File: addon/components/uni-input-price.js [hunk 5]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
...
+      }
+
+      return value;
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]: Do you need to return on the set?

File: addon/components/uni-input-price.js [hunk 6]
```diff
@@ -19,13 +21,41 @@ export default Component.extend({
...
+    }
+  }),
+
+  _isInt(value) {
```
Comment (AndreJoaquim) [inline addon/components/uni-input-price.js]: Remove this function

File: addon/templates/components/uni-input-price.hbs [hunk 7]
```diff
@@ -4,16 +4,22 @@
 
 {{input
   type=type
-  value=value
+  value=numberAsString
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-input-price.hbs]: numberAsInt

File: addon/templates/components/uni-input-price.hbs [hunk 8]
```diff
@@ -4,16 +4,22 @@
...
   max=max
   min=min
   maxlength=maxLength
   placeholder=placeholder
   focus-in=(action "onFocusIn")
-  key-down=(action "onChange")}}
+  key-up=(action "onChange")}}
+
+{{#if per}}
+
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-input-price.hbs]: Remove empty line

File: addon/templates/components/uni-input-price.hbs [hunk 9]
```diff
@@ -4,16 +4,22 @@
...
+    <span>{{currency}} / {{per}}</span>
+  {{else}}
+    <span>/ {{per}}</span>
+  {{/if}}
 
-{{#if isRightSideCurrency}}
-  <span>{{currency}} / {{per}}</span>
 {{else}}
-  <span>/ {{per}}</span>
+  <span>{{currency}}</span>
```
Comment (AndreJoaquim) [inline addon/templates/components/uni-input-price.hbs]: Remove empty spaces at EOL

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]

**Review by AndreJoaquim — APPROVED** [review]
Good work! 👍

File: addon/components/uni-input-price.js [hunk 10]
```diff
@@ -16,16 +18,40 @@ export default Component.extend({
...
+   */
+  numberAsInt: computed('value', {
+    get() {
+      return isEmpty(this.get('value')) ? '' : `${this.get('value')}`;
+    },
+
+    set(_, value) {
+      if (this.get('type') !== INPUT_TYPE_NUMBER) {
+        this.set('value', value);
+        return value;
```
Comment (montoias) [inline addon/components/uni-input-price.js]: new line

File: addon/components/uni-input-price.js [hunk 11]
```diff
@@ -16,16 +18,40 @@ export default Component.extend({
...
+      }
+
+      let newValue = isNaN(value) ? value : parseInt(value);
+      this.set('value', newValue);
+      return newValue;
```
Comment (montoias) [inline addon/components/uni-input-price.js]: new line

File: addon/components/uni-input-price.js [hunk 12]
```diff
@@ -16,16 +18,40 @@ export default Component.extend({
...
+  type: INPUT_TYPE_NUMBER,
   options: null,
 
+  /**
+   * Converts a value to an integer when able, or leaves it as a string
+   * @public
+   */
+  numberAsInt: computed('value', {
+    get() {
+      return isEmpty(this.get('value')) ? '' : `${this.get('value')}`;
```
Comment (montoias) [inline addon/components/uni-input-price.js]: Make sure the 'value' variable has a default value of empty string.

```js
get() {
   return this.get('value');
}
```

**Review by montoias — CHANGES_REQUESTED** [review]

**Review by montoias — APPROVED** [review]

**Pull request merged**